opm/autodiff/NewtonSolver.hpp

```diff
@@ -66,7 +66,7 @@ namespace Opm {
         /// \param[in]      param   parameters controlling nonlinear Newton process
         /// \param[in, out] model   physical simulation model
         explicit NewtonSolver(const SolverParameters& param,
-                              PhysicalModel& model);
+                              std::shared_ptr<PhysicalModel> model);
```

Comment:
> object lifecycle management was the whole reason for this change

Well, in that case you're probably fine although a comment to that effect would not og amiss.

> the performance implications mentioned by Sutter are not even measurable

I didn't care too much about that (on this occasion).  I was more concerned about the correctness implications of a smart pointer parameter.